lib/asciidoctor/substitutors.rb

```diff
@@ -971,11 +971,14 @@ def highlight_source source, process_callouts
   #
   # source - The String source.
   # spec   - The lines specifier (e.g., "1-5, !2, 10" or "1..5;!2;10")
+  # start  - The start attribute
   #
   # Returns an [Array] of unique, sorted line numbers.
-  def resolve_lines_to_highlight source, spec
+  def resolve_lines_to_highlight source, spec, start
     lines = []
     spec = spec.delete ' ' if spec.include? ' '
+    # If a start value is set, then the lines to highlight specified by the provided spec should be relative to the start value.
+    offset = start ? start - 1 : 0
```

Comment:
Let's keep the value nil if the start is 1. That why we avoid an unnecessary calculation per entry.

Reply:
I think it makes more sense to remove the offset from each entry after the lines have been resolved. That reduces it to a single place in the code.

```
start == 1 ? lines.sort : (lines.sort.map {|it| it - start + 1 })
```

Reply:
That's a good idea.
I wonder if we should check that we won't produce negative numbers if start is greater than highlight?

Reply:
I don't think we should worry about negative numbers. They will effectively be ignored anyway by the syntax highlighter.

Reply:
Indeed, I've added a test to make sure that the code is robust.